pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,336 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	if switchdevRunScript == nil {
+		data, err := ioutil.ReadFile(configuresSwitchdevScript)
+		if err != nil {
+			return err
+		}
+		if err := yaml.Unmarshal(data, &switchdevRunScript); err != nil {
+			return err
+		}
+	}
+
+	if networkManagerService == nil {
+		data, err := ioutil.ReadFile(networkManagerUnitFile)
+		if err != nil {
+			return err
+		}
+		var networkManagerServiceContent McoServiceInjection
+		if err := yaml.Unmarshal(data, &networkManagerServiceContent); err != nil {
+			return err
+		}
+		networkManagerService = &service.Service{
+			Name:    networkManagerServiceContent.Name,
+			Path:    systemdDir + networkManagerServiceContent.Name,
+			Content: networkManagerServiceContent.Dropins[0].Contents,
+		}
+	}
+
+	if openVSwitchService == nil {
+		data, err := ioutil.ReadFile(ovsUnitFile)
+		if err != nil {
+			return err
+		}
+		var ovsServiceContent McoServiceInjection
+		if err := yaml.Unmarshal(data, &ovsServiceContent); err != nil {
+			return err
+		}
+		openVSwitchService = &service.Service{
+			Name:    ovsServiceContent.Name,
+			Path:    systemdDir + ovsServiceContent.Name,
+			Content: ovsServiceContent.Dropins[0].Contents,
+		}
+	}
+
+	return nil
+}
+
+func handleServices() (*k8sUpdateTarget, error) {
+	updateTarget := &k8sUpdateTarget{}
+
+	// Check switchdev service
+	swdService, err := serviceManager.ReadService(switchdevService.Path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		// service not exists
+		updateTarget.switchdevService = true
+	} else {
+		needChange, err := service.CompareServices(swdService, switchdevService)
+		if err != nil {
+			return nil, err
+		}
+		updateTarget.switchdevService = needChange
+	}
+
+	// Check switchdev run script
+	data, err := ioutil.ReadFile(path.Join(chroot, switchdevRunScript.Path))
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		updateTarget.switchdevScript = true
+	} else if string(data) != switchdevRunScript.Contents.Inline {
+		updateTarget.switchdevScript = true
+	}
+
+	// Check Network Manager
+	needChange, err := checkServiceNeedUpdate(networkManagerService)
+	if err != nil {
+		return nil, err
+	}
+	updateTarget.networkManager = needChange
+
+	// Check OpenVSwitch
+	needChange, err = checkServiceNeedUpdate(openVSwitchService)
+	if err != nil {
+		return nil, err
+	}
+	updateTarget.openVSwitch = needChange
+
+	return updateTarget, nil
+}
+
+func checkServiceNeedUpdate(serviceObj *service.Service) (bool, error) {
+	systemService, err := serviceManager.ReadService(serviceObj.Path)
+	if err != nil && !os.IsNotExist(err) {
+		return false, err
+	}
+	if systemService != nil {
+		needChange, err := service.CompareServices(systemService, serviceObj)
+		if err != nil {
+			return false, err
+		}
+		return needChange, nil
+	}
+
+	return false, nil
+}
+
+func updateService(serviceObj *service.Service) error {
+	systemService, err := serviceManager.ReadService(serviceObj.Path)
```

Comment:
if you will move the compare here you won't need to check if that service need update.
Just do it here.

Reply:
`updateService` is called from `Apply` when there is change needed, the compare is running from `OnNodeChange` which check if there is change needed to be applied